src/core/window.c

```diff
@@ -3652,6 +3671,32 @@ meta_window_update_monitor (MetaWindow *window,
     }
 }
 
+
+void
+meta_window_expand_allocated_geometry (MetaWindow *window,
+                                       int        width,
+                                       int        height)
+{
+    MetaRectangle next = {
+        .x = window->currently_allocating_rect.x,
+        .y = window->currently_allocating_rect.y,
+        .width = (width < window->currently_allocating_rect.width ?
+                  window->currently_allocating_rect.width : width),
+        .height = (height < window->currently_allocating_rect.height ?
+                   window->currently_allocating_rect.height : height)
+    };
+
+    window->currently_allocating_rect = next;
+}
+
+void
+meta_window_get_minimum_size_allocation (MetaWindow *window,
```

Comment:
I'd like to keep this function around as it "completes" the API, similar to the way that gtk_widget_size_allocate works.